### PR TITLE
Allow to use cmake/ctest from PATH

### PR DESCRIPTION
**Pull request by diorcety**

### CODE REVIEW

**Comment by cowwoc** [comment]
I am fine with adding this feature, but the implementation details need to be cleaned up. Please try to keep paths inside a `File` or `Path` instead of a `String` for as long as possible.

**Comment by diorcety** [comment]
This is what i'm doing. If we want to use cmake from PATH, putting cmake in File like this: new File("cmake") it's not correct(it would mean cmake from the current directory). Added to this that i can't factorize getAbsolutePath() in each cases.

File: cmake-maven-plugin/src/main/java/com/googlecode/cmakemavenproject/CompileMojo.java [hunk 1]
```diff
@@ -125,12 +125,12 @@ public void execute()
 
 			// We assume that someone else has extracted the cmake binaries beforehand (e.g. the
 			// "generate" mojo).
-			File cmakeFile = downloadBinaries ? new File(project.getBuild().getDirectory(),
-				"dependency/cmake/bin/cmake") : new File(cmakeRootDir + "/" + cmakeChildDir);
+			String cmakePath = downloadBinaries ? new File(project.getBuild().getDirectory(),
+				"dependency/cmake/bin/cmake").getAbsolutePath() : cmakeRootDir + cmakeChildDir;
```
Comment (stephen-webb) [inline cmake-maven-plugin/src/main/java/com/googlecode/cmakemavenproject/CompileMojo.java]: To work on windows, this should be
String cmakePath = new File(downloadBinaries ? project.getBuild().getDirectory(),
				"dependency/cmake") : cmakeRootDir).getAbsolutePath() + cmakeChildDir;

**Comment by cowwoc** [comment]
This PR has been superseeded by https://github.com/cmake-maven-project/cmake-maven-project/pull/31

**Pull request closed without merging**